File: test_python_project/src/export_module/actions/perform_export.py
from utils.base_action import BaseBusinessAction
from core_module.actions.get_tasks_list import GetTasksList
from ..tasks.run_exports import task_ExportAllTasks
from core_module.tasks.run_todo_sync import task_RunTodoSync

class PerformExport(BaseBusinessAction):
    def execute(
        self,
        user_id: int,
    ) -> list[int]:
        list = GetTasksList().execute(user_id=user_id)
        task_RunTodoSync.si().apply_async()
        task_ExportAllTasks.si().apply_async()
        return list

File: test_typescript_project/src/export_module/actions/perform_export.ts
import { BaseBusinessAction } from "@/utils/base_action";
import type { TaskDTO } from "@/export_module/serializers/task_serializer";

export type ExportResult = Readonly<{
  exported: number;
  destination: "s3" | "local" | "noop";
}>;

export class PerformExport extends BaseBusinessAction<{ tasks: TaskDTO[] }, ExportResult> {
  execute(input: { tasks: TaskDTO[] }): ExportResult {
    return { exported: input.tasks.length, destination: "noop" };
  }
}

